src/runmode-dpdk.c

```diff
@@ -355,8 +356,32 @@ static void ConfigSetIface(DPDKIfaceConfig *iconf, const char *entry_str)
 static int ConfigSetThreads(DPDKIfaceConfig *iconf, const char *entry_str)
 {
     SCEnter();
-    const char *active_runmode = RunmodeGetActive();
+    static int32_t remaining_auto_cpus = -1;
+    static uint32_t total_cpus = 0;
+    if (!threading_set_cpu_affinity) {
+        SCLogError("DPDK runmode requires configured thread affinity");
+        SCReturnInt(-EINVAL);
+    }
+
+    ThreadsAffinityType *wtaf = GetAffinityTypeFromName("worker-cpu-set");
```

Comment:
can wtaf be NULL?

Reply:
The check on line 361 should have avoided that but I guess it is possible to have set threading.set-affinity to true but have commented out manager/worker threads? I'll need to check...

Reply:
GetAffinityTypeFromName returns an initialized wtaf/mtaf even if it is not present in the config. So it does not return NULL value if queried for worker/management CPU sets. However, I've added also NULL checks now.